### PR TITLE
Added two DoubleSuppliers to Java Command

### PR DESCRIPTION
**Pull request by cpapplefamily**
I added the fields in the basic Command to allow users to add two DoubleSuppliers as parameters to the command. This one issue was a significant work flow tripping point mostly for the arguably #1 command. DriveWithJoysticks. 

There is alot of work that should be done like adding the ability to name the Parameters. Currently it forces the default m_doublesupplier1 and m_doublesupplier2. Maybe a popup will be needed here to select the joystick and axis along with the variable name intended like speed and turn.

### CODE REVIEW

**Comment by cpapplefamily** [comment]
I'll look deeper into the fails but have a question. I did press on with trying to create a supplier type that only populates under the commands like all the other objects do under subsystem, io, ext.. And I did have success with replacing my DoubleSupplier 1 that took a Joystick and now only except my supplier that encompasses the available joysticks and sets the axis.

My issue is does the property field when an object is selected keep an instance of the object that we can access its properties or does it only store the name string of the object,  because now when I am creating the button assignments I can't seem to drill into the nested objects variables. 
$commad.getproperty("DoubleSupplier 1").? Now get properties from that object witch contains two properties 
 joystick name and the axis #

**Comment by sciencewhiz** [comment]
I'm not sure I'd want to go as far as having RobotBuilder providing the logic between the joystick and the doublesupplier command parameter. Rather, I would see RobotBuilder providing the template of a method that returns a double, and instantiate the command with that method reference as a double supplier and the user would fill in the logic to make that method return the value of the joystick they want. 

This could be mechanized by having a DoubleSupplier Palette item that could be added to the Subystem or OI. and a DoubleSupplier parameter that could be referenced to that Palette item, like how parameter presets can be hooked up to subsystem constants.

**Comment by sciencewhiz** [comment]
If all you have is the name, you can use the helper getByName method to get the object.

**Comment by cpapplefamily** [comment]
> I'm not sure I'd want to go as far as having RobotBuilder providing the logic between the joystick and the doublesupplier command parameter. Rather, I would see RobotBuilder providing the template of a method that returns a double, and instantiate the command with that method reference as a double supplier and the user would fill in the logic to make that method return the value of the joystick they want.
> 
> This could be mechanized by having a DoubleSupplier Palette item that could be added to the Subystem or OI. and a DoubleSupplier parameter that could be referenced to that Palette item, like how parameter presets can be hooked up to subsystem constants.

I can see your point with not having Robotbuilder make all the logic but to respectfully argue against that with tow arguments: 

- Joysticks already sort of do this with the button assignments, 
- With the proposal of having the doublesupplier stubbed in with a template returns me to exsactly where we are today. When I created a New Command and I go into the generated code to add all my DoubleSupplier variables, if I regenerate the code with Robotbuilder all my edits are overwritten. Sure I could delete the //Start.. //End.. blocks but should I?

I created a new branch in my fork called [add_DoubleSupplier_Type](https://github.com/cpapplefamily/RobotBuilder/tree/add_DoubleSupplier_Type) that when using your hint to use a helper now allowed me to create a supplier type that can currently be created by Right Clicking a Command and adding it to that command for selection in the Double Supplier 1 and 2 parameters fields. It may not be ready for Primetime and a Icon could or should be created. Should this Icon be a new section or just a component of the command pallet.

File: src/main/resources/export/java/Command-constructors.java [hunk 1]
```diff
@@ -3,19 +3,21 @@
 #set($len = $params.size() - 2)
 #set($last = $len + 1)
 
+
+
 #if ($command.getProperty("Requires").getValue() != "None")
 
 #if( $params.size() > 0 )
-    public #class($command.name)(#if( $len >= 0 )#foreach($i in [0..$len])#param_declaration_java($params.get($i)), #end#end#param_declaration_java($params.get($last)), #class($command.getProperty("Requires").getValue()) subsystem) {
+    public #class($command.name)(#if($command.getProperty("DoubleSupplier 1").getValue() != "None")DoubleSupplier doublesupplier1, #end#if($command.getProperty("DoubleSupplier 2").getValue() != "None")DoubleSupplier doublesupplier2, #end#if( $len >= 0 )#foreach($i in [0..$len])#param_declaration_java($params.get($i)), #end#end#param_declaration_java($params.get($last)), #class($command.getProperty("Requires").getValue()) subsystem) {
```
Comment (aimiebell) [inline src/main/resources/export/java/Command-constructors.java:11]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.

File: src/main/resources/export/java/Command-variable-declarations.java [hunk 2]
```diff
@@ -5,4 +5,7 @@
 #set($params = $command.getProperty("Parameters").getValue())
 #foreach($param in $params)
     private $param.getType() m_$param.getName();
-#end ## variable declaration
\ No newline at end of file
+#end ## variable declaration
+
```
Comment (aimiebell) [inline src/main/resources/export/java/Command-variable-declarations.java:9]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.

File: src/main/resources/export/java/Command-variable-setting.java [hunk 3]
```diff
@@ -2,4 +2,7 @@
 #set($params = $command.getProperty("Parameters").getValue())
 #foreach($param in $params)
         m_$param.getName() = $param.getName();
-#end
\ No newline at end of file
+#end
+#if($command.getProperty("DoubleSupplier 1").getValue() != "None")m_doublesupplier1 = doublesupplier1; #end
```
Comment (aimiebell) [inline src/main/resources/export/java/Command-variable-setting.java:6]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.

File: src/main/resources/export/java/Command.java [hunk 4]
```diff
@@ -5,7 +5,7 @@
 #set($params = $command.getProperty("Parameters").getValue())
```
Comment (aimiebell) [inline src/main/resources/export/java/Command.java:5]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.

**Comment by cpapplefamily** [comment]
I took a different angle on this in PR #379

**Comment by cpapplefamily** [comment]
this can be closed if #474 is adopted

**Pull request closed without merging**